File: PhoneRepairEnd/src/main/java/com/hnust/pojo/Management.java
package com.hnust.pojo;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
@Data
@TableName(value = "yjx_repair_management")  // 指定数据库表名
@Builder  // 构造器
@AllArgsConstructor  // 全参构造函数
@NoArgsConstructor  // 无参构造函数
public class Management {
    @TableId(type = IdType.AUTO)
    private Integer repairId; //维修id
    private Integer repairRequestId; //订单id
    private Integer technicianId; //维修人员id
    private Double repairPrice; //维修价格
    private String paymentStatus;//支付状态
    private String repairNotes; //维修备注
    private LocalDateTime createdAt; //创建时间
    private LocalDateTime updatedAt; //更新时间

    // 2. 新增非数据库字段（从关联表获取，前端必需）
    @TableField(exist = false)  // 标记：该字段不在 yjx_repair_management 表中
    private String phoneModel;  // 手机型号（来自 yjx_repair_request.phone_model）

    @TableField(exist = false)
    private String statusName;  // 维修状态名称（来自 yjx_repair_request.request_status，转文字）

    @TableField(exist = false)
    private String userName;    // 订单用户名（来自 yjx_user.user_name
}

File: PhoneRepairFrontV3/src/types/management.ts
/** 维修管理记录 */
export interface ManagementRecord {
  repairId: number
  repairRequestId: number
  technicianId: number
  repairPrice: number
  paymentStatus: string
  repairNotes: string
  createdAt: string
  phoneModel: string
  statusName: string
  userName: string
}

/** 创建维修管理参数 */
export interface ManagementCreateParams {
  repairRequestId: number
  repairNotes: string
  technicianId: number
}

/** 更新维修管理参数 */
export interface ManagementUpdateParams {
  repairId: number
  repairPrice?: number
  paymentStatus?: string
  repairNotes?: string
  technicianId: number
}
